rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -66,22 +74,98 @@ public String getDescription() {
         return "Merge a YAML snippet with an existing YAML document.";
     }
 
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        if (StringUtils.isNullOrEmpty(filterYaml)) {
+            return null;
+        }
+
+        Yaml.Documents filter = new YamlParser().parse(filterYaml).get(0);
+        Set<YamlSearchResult> searchResults = new HashSet<>();
+
+        return new YamlVisitor<ExecutionContext>() {
+            @Override
+            public Yaml visitSequence(Yaml.Sequence sequence, ExecutionContext ctx) {
+                new ContainsTree<>(sequence, filter, key).visit(sequence, searchResults, getCursor());
+                if (doesNotMatch(searchResults)) {
+                    return sequence;
+                } else {
+                    return sequence.withMarkers(sequence.getMarkers().addIfAbsent(ContainsTree.MATCHES));
+                }
+            }
+
+            @Override
+            public Yaml visitDocuments(Yaml.Documents documents, ExecutionContext ctx) {
+                new ContainsTree<>(documents, filter, key).visit(documents, searchResults, getCursor());
+                if (doesNotMatch(searchResults)) {
+                    return documents;
+                } else {
+                    return documents.withMarkers(documents.getMarkers().addIfAbsent(ContainsTree.MATCHES));
+                }
+            }
+        };
+    }
+
     @Override
     protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        Yaml.Documents incoming = new YamlParser().parse(yaml).get(0);
+
+        if ("/".equals(key)) {
+            return new YamlVisitor<ExecutionContext>() {
+                @Override
+                public Yaml visitDocuments(Yaml.Documents documents, ExecutionContext ctx) {
+                    Yaml.Documents docs = (Yaml.Documents) new MergeYamlVisitor(documents, incoming).visit(documents, ctx, getCursor());
+                    doAfterVisit(new AutoFormatVisitor<>(null));
```

Comment:
We generally want to make the most minimal possible changes. This will reformat all of the document(s), including those parts unrelated to the merge operation itself. Ideally the only bits that would ever be AutoFormatted would be those you added

Reply:
How do we do that with a deep merge? Can we autoformat only part of a tree?

Reply:
Yes, you can autoformat only part of the tree. That's what I was showing you yesterday with the three argument visit method.